wordpress-mobile/WordPress-iOS
# Editing: Show notice when switching editor mode

Fixes #17494

This pull-request adds a subtle notice at the top of the editor when switching between HTML and Visual modes.

**Note** This notice is only shown on the Gutenberg editor.

The idea started in [a discussion](https://github.com/wordpress-mobile/WordPress-Android/issues/11819#issuecomment-969670515) on an issue for the WordPress-Android app where the proposed solution was to show a notice to the user when switching the editor mode, instead of a snackbar.

It was suggested to add this notice also to the iOS app to have both apps aligned.

## To test:
1. From My Site tap the Posts button or scroll down and tap Posts from the list
2. In post cards mode tap the Edit button / In post list mode tap the post
3. In the editor, tap the 3 dots in the upper right corner
4. Tap "Switch HTML Mode" (first button)
5. **Verify** that a compact notice appears at the top of the editor
6. Tap again the 3 dots in the upper right corner
7. Tap "Switch Visual Mode" (first button)
8. **Verify** that a compact notice appears at the top of the editor

## Preview
<details>
  <summary>Switching to HTML mode</summary>

| Before | After |
| --- | --- |
| ![1_ios_switch_to_html_before](https://user-images.githubusercontent.com/4588074/142916696-5eef247c-1f9e-4b8d-80b6-310eeff27136.png) | ![1_ios_switch_to_html_after](https://user-images.githubusercontent.com/4588074/142916714-7133089c-2f9b-421f-95cd-099658180f89.png) |

</details>

<details>
  <summary>Switching to Visual mode</summary>

| Before | After |
| --- | --- |
| ![2_ios_switch_to_visual_before](https://user-images.githubusercontent.com/4588074/142916968-be253411-4406-459d-8984-77621bf27c1d.png) | ![2_ios_switch_to_visual_after](https://user-images.githubusercontent.com/4588074/142916980-f740614c-3f7d-476f-b611-ad034ea14f07.png) |

</details>

<details>
  <summary>Video Preview</summary>

https://user-images.githubusercontent.com/4588074/142916268-590f610b-d4f6-4685-9fb9-d61ae2d2aa4d.mov
</details>


## Regression Notes
1. Potential unintended areas of impact
  None because the notice is only shown when switching the editor mode in the Gutenberg editor.  The changes do not affect any other area of the app.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
  Manual testing.

3. What automated tests I added (or what prevented me from doing so)
  None, since this change does not warrant such tests.

PR submission checklist:
